ttnn/cpp/ttnn/operations/ccl/ccl_op_fusion.cpp

```diff
@@ -12,11 +12,13 @@ namespace ccl {
 
 void AllGatherFusedOpSignaler::init_fused_op(
     const std::vector<CoreCoord>& fused_op_receiver_cores_noc,
-    const std::vector<uint32_t>& fused_op_receiver_signal_semaphores
+    const std::vector<uint32_t>& fused_op_receiver_signal_semaphores,
+    bool mcast_fused_op_cores
 ) {
     this->fused_op_receiver_cores_noc = fused_op_receiver_cores_noc;
     this->fused_op_receiver_signal_semaphores = fused_op_receiver_signal_semaphores;
     this->num_fused_op_cores_to_signal = fused_op_receiver_cores_noc.size();
+    this->mcast_fused_op_cores = mcast_fused_op_cores;
```

Comment:
Just confirming understanding here. This is to enable a mcast to signal the consumer op worker rectangle instead of unicasting individually?

Reply:
When the mcast mode is true, you signal all the cores in a for loop, whenever all gather want's to signal. So it's not really dependent on the shape of the op worker core grid. 

When the mcast mode is false, you signal only one core, whenever all gather want's to signal.